powerdns_sync/lib/Atomia/DNS/PowerDNSSyncer.pm

```diff
@@ -149,11 +154,8 @@ sub reload_updated_zones {
 
 				my $zone_name = $zone->{"name"};
 				my $zone_status = 'active';
-
-				foreach my $zonestatusitem (@zonestatusbatch) {
-					if ($zone_name eq $zonestatusitem->{zonename}) {
-						$zone_status = $zonestatusitem->{zonestatus};
-					}
+				if(exists($zone_status_hash->{$zone_name}) && $zone_status_hash->{$zone_name} eq "suspended"){
```

Comment:
How will this piece of code `exists($zone_status_hash->{$zone_name}` behave if zone_status_hash is empty?